From dae0d5e161cedfcdd7107140ac7dbc4116b3e591 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Thu, 30 Oct 2014 17:31:57 -0700
Subject: [PATCH 004/184] MIPS: OCTEON: Handle NULL of_node in
 of_node_to_nid().

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 608322637716f0a0f6b8a13470af54f1494f3509
Description:

If there is no of_node, just say it is on node 0.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from octeon-linux-kernel-patches-SDK-3.1.2-release]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/cavium-octeon/octeon-numa.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/mips/cavium-octeon/octeon-numa.c b/arch/mips/cavium-octeon/octeon-numa.c
index 2751e5b..6e4e7c7 100644
--- a/arch/mips/cavium-octeon/octeon-numa.c
+++ b/arch/mips/cavium-octeon/octeon-numa.c
@@ -105,8 +105,12 @@ void __init mem_init(void)
 int of_node_to_nid(struct device_node *np)
 {
 	int ret = 0;
-	struct device_node *node = of_node_get(np);
+	struct device_node *node;
 
+	if (!np)
+		return 0;
+
+	node = of_node_get(np);
 	do {
 		if (strcmp("soc", node->name) == 0) {
 			int rc;
-- 
1.9.1

